spec/formatting.md

```diff
@@ -600,227 +608,77 @@ First, resolve the values of each _selector_:
-   resulting in a list « `one`, `1`, `*` » of variants.
+Next, using `res`:
 
-3. Sorting the variants:<br>
-   The list `sortable` is first set with the variants in their source order
-   and scores determined by the selector key order:<br>
-   « ( 1, `one` ), ( 0, `1` ), ( 2, `*` ) »<br>
-   This is then sorted as:<br>
-   « ( 0, `1` ), ( 1, `one` ), ( 2, `*` ) »<br>
+1. Let `bestVariant` be `UNSET`.
+1. For each _variant_ `var` of the message, in source order:
+   1. Let `keys` be the keys of `var`.
+   1. Let `match` be SelectorsMatch(res, keys).
+   1. If `match` is false:
+      1. Continue the loop.
+   1. If `bestVariant` is `UNSET`.
+      1. Set `bestVariant` to `var`.
+   1. Else:
+      1. Let `bestVariantKeys` be the keys of `bestVariant`.
+      1. If SelectorsCompare(res, keys, bestVariantKeys) is `BETTER`:
+         1. Set `bestVariant` to `var`.
+1. Select the _pattern_ of `bestVariant`.
+
+#### SelectorsMatch
+
+SelectorsMatch(selectors, keys) is defined as follows, where
+`selectors` is a list of resolved values
+and `keys` is a list of keys:
+
+1. Let `i` be 0.
+1. For each key `key` in `keys`:
+   1. If `key` is the catch-all key `'*'`
+      1. Set `i` to `i` + 1.
+      1. Continue the loop.
+   1. Let `k` be the _resolved value_ of `key` in Unicode Normalization Form C [\[UAX#15\]](https://www.unicode.org/reports/tr15).
+   1. Let `sel` be the `i`th element of `selectors`.
+   1. If Match(sel, k) is false:
+      1. Return false.
+   1. Set `i` to `i` + 1.
+1. Return true.
+
+#### SelectorsCompare
+
+SelectorsCompare(selectors, keys1, keys2) is defined as follows, where
+`selectors` is a list of resolved values
+and `keys1` and `keys2` are lists of keys.
+
+1. Let `result` be `SAME`.
+1. Let `i` be 0.
+1. For each key in `keys1`:
+   1. Let `key1` be the `i`th element of `keys1`.
+   1. Let `key2` be the `i`th element of `keys2`.
+   1. If `key1` is the catch-all key `'*'` and `key2` is not the catch-all key:
+      1. Return `WORSE`.
+   1. If `key1` is not the catch-all key `'*'` and `key2` is the catch-all key:
+      1. Return `BETTER`.
+   1. If `key1` and `key2` are both the catch-all key `'*'`
+      1. Set `i` to `i + 1`.
+      1. Continue the loop.
+   1. Let `k1` be the _resolved value_ of `key1` in Unicode Normalization Form C [\[UAX#15\]](https://www.unicode.org/reports/tr15).
+   1. Let `k2` be the _resolved value_ of `key2` in Unicode Normalization Form C [\[UAX#15\]](https://www.unicode.org/reports/tr15).
+   1. Let `sel` be the `i`th element of `selectors`.
+   1. Set `result` to Compare(sel, k1, k2).
+   1. If `result` is `SAME`:
+      1. Set `i` to `i + 1`.
+      1. Continue the loop.
+   1. Else:
+      1. Return `result`.
+1. Return `result`.
 
-4. The pattern `Exact match for {$count}` of the most preferred `1` variant is selected.
+For examples of how the algorithms work, see [the appendix](#appendix-non-normative-examples).
```

Comment:
Given that we do have `appendices.md`, shouldn't the appendix of selection examples go there instead?

Reply:
Changed in [4b577a6](https://github.com/unicode-org/message-format-wg/pull/1080/commits/4b577a69ceaaff0cb1b1c856281df682ec22930d)